pkg/proxy/config/api_test.go

```diff
@@ -174,24 +165,16 @@ func TestNewEndpointsSourceApi_UpdatesAndMultipleEndpoints(t *testing.T) {
 
 	ch := make(chan EndpointsUpdate)
 
-	cache.NewReflector(lw, &api.Endpoints{}, NewEndpointsStore(nil, ch), 30*time.Second).Run()
-
-	got, ok := <-ch
-	if !ok {
-		t.Errorf("Unable to read from channel when expected")
-	}
-	expected := EndpointsUpdate{Op: SET, Endpoints: []api.Endpoints{}}
-	if !apiequality.Semantic.DeepEqual(expected, got) {
-		t.Errorf("Expected %#v; Got %#v", expected, got)
-	}
+	endpointsController := NewEndpointsController(lw, 30*time.Second, ch)
+	go endpointsController.Run(wait.NeverStop)
```

Comment:
> So this used to be meaningful, in that the first update was a "clear" if there were no endpoints

Sorry - don't understand this. Can you please clarify?

But anyway, I agree that previously "SET" was handled differently than "ADD", because:
- ADD was used only in unit tests (i.e. production code was never issuing ADD) - it was always taking all contents from the underlying store and calling "SET" (meaning "replace") from those contents (no matter if it was add, update or remove).

However, with the controller framework, "REPLACE" is not an external concept.
If we are doing relist, controller internally is handling it - it is computing the difference and sending appropriate Add, Update and Remove calls if needed to match the desired state.
The same for resync (though since resync is called from in-memory state, there are pretty much only update handlers called).

That means, that since those are internal details of controller framework, we don't have a good was for forcing it. Also, those are already tested at the level of Informer, so testing framework internals here doesn't sound like a good idea to me.

But if you don't agree, can you please clarify what exactly you want to be tested, since I'm not sure I understood it correctly.